src/reducers/user.reducer.js

```diff
@@ -5,15 +5,17 @@ import {
     RECEIVE_ROLES,
 } from '../actions/user.actions';
 
-const userReducer = (state = {}, action) => {
+const userReducer = (state = { isAuthorized: false }, action) => {
+    const { payload } = action;
+
     switch (action.type) {
         case SAVE_USER:
         case REQUEST_USER:
             return { ...state, isFetching: true };
         case RECEIVE_USER:
-            return { ...state, account: action.payload, isFetching: false };
+            return { ...state, account: payload, isFetching: false, isAuthorized: Boolean(payload._id) };
```

Comment:
I believe we shouldn't add more flags to state. All we need - user object.
The problem should be fixed on both client and server sides: server should respond with 401 on GET /user and client should process it in correct way